### PR TITLE
Dockerfile: Add init-container.sh to cilium image

### PR DESCRIPTION
**Pull request by nirmoy**
This will allows us to make use of same cilium
image instead of cilium-init image for clean-cilium-state
init container

Todo after this PR is gets merged:
1 Recreate cilium image
2 create a another PR to change `__CILIUM_INIT_VERSION__` to `__CILIUM_ERSION__` in `examples/kubernetes/templates/v1`

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8489)

### CODE REVIEW

File: contrib/packaging/docker/Dockerfile.runtime [hunk 1]
```diff
@@ -1,100 +0,0 @@
-#
```
Comment (aanm) [inline contrib/packaging/docker/Dockerfile.runtime]: This should not have been removed :/

**Review by aanm — CHANGES_REQUESTED** [review]

File: contrib/packaging/docker/Dockerfile.runtime (hunk 1 above)
Comment (nirmoy) [reply contrib/packaging/docker/Dockerfile.runtime]: @aanm :man_facepalming:

**Comment by nirmoy** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
@nirmoy you can as well change the k8s descriptors to point to this image instead

**Review by ianvernon — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Pull request merged**